shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -890,13 +900,18 @@ private void finishFrame(boolean isFrameRenderedUsingImageReaders) {
       final int viewId = platformViewParent.keyAt(i);
       final View parentView = platformViewParent.get(viewId);
 
-      // Show platform views only if the surfaces have images available in this frame,
-      // and if the platform view is rendered in this frame.
+      // This should only show platform views that are rendered in this frame and either:
+      //  1. Surface has images available in this frame or,
+      //  2. Surface does not have images available in this frame because the render surface should
+      // not be an ImageView.
+      //
       // The platform view is appended to a mutator view.
       //
       // Otherwise, hide the platform view, but don't remove it from the view hierarchy yet as
       // they are removed when the framework diposes the platform view widget.
-      if (isFrameRenderedUsingImageReaders && currentFrameUsedPlatformViewIds.contains(viewId)) {
+      if (!currentFrameUsedPlatformViewIds.contains(viewId)) {
+        parentView.setVisibility(View.GONE);
+      } else if (isFrameRenderedUsingImageReaders || convertRenderSurfaceDisabled) {
```

Comment:
I'd suggest making it a bit more readable. One rule of thumb is usually to prefer positive assertions as opposed to negative assertions. e.g.  `if (currentFrameUsedPlatformViewIds.contains(viewId)) {} else { }` is better than `if (!currentFrameUsedPlatformViewIds.contains(viewId)) {} else { }`